Kudu.Services/Diagnostics/XenonProcessController.cs

```diff
@@ -0,0 +1,184 @@
+            }
+        }
+
+        [HttpGet]
+        public HttpResponseMessage GetAllProcesses(bool allUsers = false)
+        {
+            var requestUri = Request.GetRequestUri(_settings.GetUseOriginalHostForReference()).GetLeftPart(UriPartial.Path).TrimEnd('/');
+            using (_tracer.Step("XenonProcessController.GetAllProcesses"))
+            {
+                return ForwardToContainer($"");
+            }
+        }
+
+        [HttpGet]
+        public HttpResponseMessage GetProcess(int id)
+        {
+            using (_tracer.Step("XenonProcessController.GetProcess"))
+            {
+                return ForwardToContainer($"{id}");
+            }
+        }
+
+        [HttpDelete, HttpPost]
+        public HttpResponseMessage KillProcess(int id)
+        {
+            using (_tracer.Step("XenonProcessController.KillProcess"))
+            {
+                return ForwardToContainer($"{id}");
+            }
+        }
+
+        [HttpGet]
+        public HttpResponseMessage MiniDump(int id, int dumpType = 0, string format = null)
+        {
+            using (_tracer.Step("XenonProcessController.MiniDump"))
+            {
+                return ForwardToContainer($"{id}/dump");
+            }
+        }
+
+        [HttpPost]
+        public HttpResponseMessage StartProfileAsync(int id, bool iisProfiling = false)
+        {
+            using (_tracer.Step("XenonProcessController.StartProfileAsync"))
+            {
+                return ForwardToContainer($"{id}/profile/start");
+            }
+        }
+
+        [HttpGet]
+        public HttpResponseMessage StopProfileAsync(int id)
+        {
+            using (_tracer.Step("XenonProcessController.StopProfileAsync"))
+            {
+                return ForwardToContainer($"{id}/profile/stop");
+            }
+        }
+
+        public HttpResponseMessage ForwardToContainer(string route)
+        {
+            using (_tracer.Step("XenonProcessController.ForwardToContainer"))
+            {
+                // Forward request to windows container
+
+                // Get the container address and the port kudu agent port
+                IDictionary environmentVariables = System.Environment.GetEnvironmentVariables();
+                if (environmentVariables.Contains("KUDU_AGENT_HOST") && environmentVariables.Contains("KUDU_AGENT_PORT")
+                    && environmentVariables.Contains("KUDU_AGENT_USR") && environmentVariables.Contains("KUDU_AGENT_PWD"))
+                {
+                    var containerAddress = environmentVariables["KUDU_AGENT_HOST"];
+                    var kuduContainerAgentPort = environmentVariables["KUDU_AGENT_PORT"];
+                    var kudu_agent_un = environmentVariables["KUDU_AGENT_USR"];
+                    var kudu_agent_pwd = environmentVariables["KUDU_AGENT_PWD"];
+
+                    string containerUrl = $"http://{containerAddress}:{kuduContainerAgentPort}/processes/{route}";
```

Comment:
Is this route already setup on the container endpoint?

Reply:
No. I am manually copying in and starting the dll that this route connects to. The dll will be in place and auto started by the time this PR is completed.